Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Search/SearchWidget.cpp

```diff
@@ -198,6 +210,29 @@ namespace AzToolsFramework
             }
         }
 
+        void SearchWidget::ToggleUnusableProductsFilter(bool checked)
+        {
+            if (!checked)
+            {
+                m_filter->RemoveFilter(FilterConstType(m_unusableProductsFilter));
+            }
+            else
+            {
+                m_filter->AddFilter(FilterConstType(m_unusableProductsFilter));
+            }
+        }
+
+        AZStd::vector<AZ::Data::AssetType> SearchWidget::BuildAssetTypeList()
+        {
+            AZStd::vector <AZ::Data::AssetType> entries = AZStd::vector<AZ::Data::AssetType>();
+            entries.push_back(AzFramework::AzslOutcomeAssetType);
+            entries.push_back(AzFramework::BufferAssetType);
+            entries.push_back(AzFramework::ImageMipChainAssetType);
+            entries.push_back(AzFramework::ModelLodAssetType);
+            entries.push_back(AzFramework::ShaderVariantAssetType);
+            return entries;
+        }
```

Comment:
This function could query the SettingsRegistry for the list of AssetTypes.

Also the vector class default constructor initializes to an empty vector there is no need to invoke the move constructor here to initialize an empty instance.
`AZStd::vector<AZ::Data::AssetType> entries;` works just as well

```suggestion
        AZStd::vector<AZ::Data::AssetType> SearchWidget::BuildAssetTypeList()
        {
            AZStd::vector<AZ::Data::AssetType> entries;
            if (AZ::SettingsRegistryInterface* settingsRegistry = AZ::SettingsRegistry::Get();
                settingsRegistry != nullptr)
            {
                auto VisitFilteredAssetTypes = [&entries](const AZ::SettingsRegistryInterface::VisitArgs& visitArgs)
                {
                    using FixedValueString = AZ::SettingsRegistryInterface::FixedValueString;
                    if (FixedValueString value{}; visitArgs.m_registry.Get(value, visitArgs.m_jsonKeyPath))
                    {
                        AZ::Data::AssetType assetType{ AZStd::string_view(value) };
                        entries.emplace_back(AZStd::move(assetType));
                    }
                };
                AZ::SettingsRegistryVisitorUtils::VisitObject(*settingsRegistry, AppendObjectFields, "/O3DE/AssetBrowser/AssetTypeUuidExcludes");
            }

            return entries;
        }
```

Reply:
Done!